<?xml version="1.0"?>

<launch>
<!--    <node pkg="hokuyo_node" type="hokuyo_node" name="hokuyo_node" output="screen">
       <param name="min_ang" value=" -2.2689" />
       <param name="max_ang" value="  2.2689" />
    </node>-->
<!--    <node pkg="hector_scanmatcher" type="hector_scanmatcher" name="hector_scanmatcher" output="screen" launch-prefix="valgrind xxtool=callgrind">-->
          <node pkg="hector_scanmatcher" type="hector_scanmatcher" name="hector_scanmatcher" output="screen" >
    <!--<node pkg="hector_scanmatcher" type="hector_scanmatcher" name="hector_scanmatcher" output="screen" launch-prefix="gdb -ex run xxargs">-->
        <param name="base_frame" value="base_stabilized" />
        <param name="odom_frame" value="nav"/>

        <!--<param name="odom_frame" value="base_stabilized"/> -->
        <param name="map_resolution" value="0.05"/>
        <param name="output_timing" value="false"/>

        <param name="use_tf_scan_transformation" value="true"/>
        <param name="use_tf_pose_start_estimate" value="false"/>
        <param name="scan_topic" value="scan"/>

<!--        <param name="map_size" value="2048" />-->
        <param name="map_size" value="2048" />
        <param name="map_start_x" value="0.125" />
        <param name="map_start_y" value="0.5" />

<!--        <param name="pub_drawings" value="true"/>
        <param name="pub_debug_output" value="true"/>-->
<!--        <param name="pub_map_odom_transform" value="true"/>-->
    </node>

    <param name="hector_scanmatcher/pub_map_odom_transform" value="true"/>
<!--    <node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster" args="0 0 0 0 0 0 map nav 100" />-->
    <node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster_1" args="0 0 0 0 0 0  nav base_footprint 100" />
    <node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster_2" args="0 0 0 0 0 0  base_footprint base_stabilized 100" />
    <node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster_3" args="0 0 0 0 0 0  base_stabilized base_link 100" />
</launch>
